File: Util/app/src/main/java/com/example/administrator/app/ShareActivity.java
package com.example.administrator.app;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.example.administrator.base.WinBaseActivity;
import com.example.administrator.util.L;
import com.example.administrator.util.ShareUtils;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * SharedPreferences 的封装
 */
public class ShareActivity extends WinBaseActivity {

    //   原生存储数据
    @BindView(R.id.btn_01)
    Button btn01;
    //    原生取出数据
    @BindView(R.id.btn_02)
    Button btn02;
    //    使用工具类存储数据
    @BindView(R.id.btn_03)
    Button btn03;
    //    使用工具类取出数据
    @BindView(R.id.btn_04)
    Button btn04;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_share);
        ButterKnife.bind(this);
    }

    @OnClick({R.id.btn_01, R.id.btn_02, R.id.btn_03, R.id.btn_04})
    public void onViewClicked(View view) {
        switch (view.getId()) {
//           原生储存数据
            case R.id.btn_01:
                SharedPreferences.Editor editor = getSharedPreferences("data", MODE_PRIVATE).edit();
                editor.putString("name", "tonjies");
                editor.putInt("age", 20);
                editor.apply();
                break;
//           原生获取数据
            case R.id.btn_02:
                SharedPreferences preferences = getSharedPreferences("data", MODE_PRIVATE);
                String name = preferences.getString("name", "");
                int age = preferences.getInt("age", 18);
                L.d("name:" + name);
                L.d("age" + age);
                break;
//            使用工具类存储数据
            case R.id.btn_03:
//               存储String类型
                ShareUtils.putString(ShareActivity.this, "name2", "tonjies");
//               存储int类型
                ShareUtils.putInt(ShareActivity.this, "age2", 21);
                break;
//               使用工具类取出数据
            case R.id.btn_04:
//               取出String类型
                final String string = ShareUtils.getString(ShareActivity.this, "name2", "没有值可以取出来");
                L.d(string);
//                取出int类型
                final int age2 = ShareUtils.getInt(ShareActivity.this, "age2", 18);
                L.d(age2 + "");
                break;
        }
    }
}

File: BaseProject/app/src/main/java/com/example/administrator/app/module/test/ShareActivity.kt
package com.example.administrator.app.module.test

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.example.administrator.app.R
import com.example.administrator.app.util.L
import com.example.administrator.app.util.ShareUtils
import kotlinx.android.synthetic.main.activity_anko.*
import org.jetbrains.anko.longToast
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.toast
import kotlinx.android.synthetic.main.activity_share.*

/**
 * sharedPreferences封装类测试
 */
class ShareActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_share)
        init()
    }


    /**
     * 虽然我们通过import kotlinx.android.synthetic.main.activity_anko.*使得不需要写控件的fndId，
     * 但是我们仍然是要写setOnClickListener(this)控件点击回调监听的
     */
    private fun init() {
        btn1.setOnClickListener(this)
        btn2.setOnClickListener(this)
        btn3.setOnClickListener(this)
        btn4.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.btn1 -> {
//              存储String类型
                ShareUtils.putString(this, "name", "tonjie")
                ShareUtils.putString(this, "age", "20")
                toast("A")
            }
//              取出String类型
            R.id.btn2 -> {
                val name = ShareUtils.getString(this, "name", "20")
                L.d("name:  " + name)
            }
            R.id.btn3 -> {
//              删除单个key
                ShareUtils.deleShare(this, "name")
            }
            R.id.btn4 -> {
//              删除所有key
                ShareUtils.deleAll(this)
            }
        }
    }
}
